### ISSUE TITLE
[Meta] Remove Windows XP components

### ISSUE THREAD

**Issue by wolfbeast** [MoonchildProductions/Pale-Moon#461]
The source tree has a number of Windows XP specific components that should be removed since we are no longer going to support it as a target. This bug is to track the specific tasks involved.
- [x] Remove XP-specific style components from /browser
- [x] Remove XP-specific style components from /toolkit
- [x] Remove `@media` queries for Windows XP themes (lunaBlue etc.)
- [x] Merge "classic" and "aero" styles since "aero" (NT >=6) is the only style left
- [x] Find and remove Windows XP specific code considerations for e.g. kernel calls

**Comment by dsp2003** [comment]
Don't you DARE. I (and thousands of other people) still use XP build. This decision will bite you hard.

**Comment by wolfbeast** [comment]
This will be dead code cleanup only. Dropping of XP support has already been decided for the next milestone. This is planned either for or after its release.
https://forum.palemoon.org/viewtopic.php?f=5&t=12216

**Comment by elypter** [comment]
why not keep it in as long as it still works or is easily patchable? is it hurting really so badly that others may use windows xp even though you think that is a bad idea? btw reactOS is also nt5

**Comment by wolfbeast** [comment]
It already no longer works in our current versions of Tycho. And yes, it does hurt development pretty badly to keep NT5 in mind. Easy patchability is currently still relatively OK but we will be rapidly moving away from NT5 cOS compliance so if you are planning to fork us for a Windows XP version or XP compatible rebuild, then now would be the time, and keeping up with possible patches that need changing for XP compatibility, lest it becomes a mountain of work to catch up.

**Comment by wolfbeast** [comment]
As for ReactOS: They have not been compatible for a while; even pre-Tycho there have been plenty of issues. Their NT5 compatibility is partial to begin with, and their development is not making any big strides. **It can't be a valid target for us at this time** because it's simply not even close to usable (trust me, I tried). ReactOS should have focused on NT6 compatibility a long time ago already but they are choosing to stay on NT5. Once more, same as for XP, if you want to fork and work on getting it usable, now would be a good time to do so.

**Comment by elypter** [comment]
but cant you still keep it in the code at least commented out. is there no way in the ide to hide comments based on tagging, so then they wont even get in your way. its not just about xp code but also about other things that might become important again. if you delete them its a huge effort to merge them back in from an older version. if you keep it you just have to fix the incompatibilities(still a lot of work but less)

**Comment by Lootyhoof** [comment]
As of Tycho the XP-specific styling should now be purged. https://github.com/MoonchildProductions/Tycho/pull/107 (main removal), with https://github.com/MoonchildProductions/Tycho/pull/116 and https://github.com/MoonchildProductions/Tycho/pull/123 for cleanup (in addition to the main reason for those pulls).

**Comment by wolfbeast** [comment]
@elypter commenting out code is bad practice in general unless you expect it to be a short-lived workaround. It makes code less readable and especially when cleaning up styles it would also require extra files to remain included which is really unmaintainable.
There's a good reason I say that if you want to take on a project like that, that you should fork now, as we *will* be removing more and more obsolete code.

MoonchildProductions/Tycho#125 would (probably) be the last to fix the first two bullet points here, so I'm marking those off at this time.

**Comment by elypter** [comment]
but its only bad practice if the ide is inferior. i dont know what software supports hiding comments based on patterns but i would be surprised if none exists. 

something like:

normal_code

//+legacy/xp_code
//void draw_xp_graphics() {
//do_stuff();
//}
other_stuff

turns into this:

normal_code

other_stuff

if your ide autohides every multiline comment that starts with //+legacy/xp_code

leaving files out of the building process should be even easier.
(there are probably other better ways of mass hiding code. thats just an example)

im not trying to tell you how you should code here. i just want to say that its not always the best way to do like everyone else does. situations change like ide's get more powerful or your project is special in a certain way which makes it worthy exploring new paths. actually your project is special in a way because you fork from a rapidly changing giant project and you are at the moment rebasing. so some of the code that you deleted from the original sources has now to be woven in again. i dont know about the internals of this project but i assume that you could have saved a little work if some of the deleted code was still where it was.

this also also increases flexibility if you for example want to be able to switch between subsystems but do not want to bake this into the binary or precompiling process

**Comment by wolfbeast** [comment]
There is no reason at all for our team to retain the code as comments. They will be useless for further development as we are not going to reintroduce this support. Keeping the code blocks "just in case" will quickly create unreadable code, as I already explained, since very regularly it won't simply be deletion of code, and interdependent functions will have their relationship changed. It's not possible to keep the previous version of that code in the actual source because it will quickly become pointless, unusable snippets that simply uncommenting won't restore the previous behavior.

This is why version control exists, and why we are using a git repository with commit history. Removing code is perfectly fine since previous code will at all times be retained in the code history, and on top, will be listed as individual commits that have all related code changes together.

You also misunderstand what most of our re-basing work involves: we are not weaving in previously deleted code. We are having to re-apply what we changed previously, and delete again what we previously deleted from the previous code base.
Commented code would not have helped in any way and in fact would have made things a lot messier.

Bottom line is that commenting out code that has no future use just for the sake of someone else _maybe_ wanting to fork this at a later point in time instead of when they should, who for some reason doesn't want to work with commit history, is bad. I won't do it.

For the last time, If people really want to keep XP compatibility in Pale Moon's future code base, they should fork **now**, and work on making code changes to restore XP compatibility (the code is no longer XP compatible as-is, some reversal is already needed, not just the style changes).

Locking this thread to contributors only at this point. It's been explained and this discussion doesn't belong on GitHub. If you want to discuss this further, go to the forum.

**Comment by wolfbeast** [comment]
Checking off the 5th point since most of the XP specific code that is obvious (and in our code, not 3rd party) has been addressed, including the dependency on DX9 for WebGL.

**Comment by wolfbeast** [comment]
Point 3 has been addressed with 16b32fa1e712adbc9d93a3e3bf8c53e833d4a635 concldugin this cleanup work.
